datus/storage/subject_tree/store.py

```
@@ -156,7 +156,13 @@ def create_node(self, parent_id: Optional[int], name: str, description: str = ""
         except sqlite3.IntegrityError as e:
             conn.rollback()
             if "UNIQUE constraint failed" in str(e):
-                raise ValueError(f"Node with name '{name}' already exists under parent {parent_id}")
+                # Race condition in parallel writes: node was created by another thread/process
+                # Return the existing node instead of raising an error
+                existing_node = self._find_child_by_name(parent_id, name)
+                if existing_node:
+                    logger.debug(f"Node '{name}' already exists under parent {parent_id}, returning existing node")
+                    return existing_node
+                raise ValueError(f"Node with name '{name}' already exists under parent {parent_id}") from e
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)